nanogpt/export_nanogpt.py

```diff
@@ -24,11 +25,29 @@
 )
 
 
+class NanoGPT(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.model = GPT.from_pretrained('gpt2') # use gpt2 weight as pretrained weight
+
+    def forward(self, idx, max_new_tokens):
+        for _ in range(20):
+            # if the sequence context is growing too long we must crop it at block_size
+            idx_cond = idx if idx.size(1) <= self.config.block_size else idx[:, -self.config.block_size:]
+            # forward the model to get the logits for the index in the sequence
+            logits, _ = self.model()(idx_cond)
+            # choose the highest probability token as the next index to continue the sequence with
+            idx_next = torch.argmax(logits).view(1, 1)
+            # append sampled index to the running sequence and continue
+            idx = torch.cat((idx, idx_next), dim=1)
+
+        return idx
+
 
 def main(args):
     # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Prep ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-    model = GPT.from_pretrained('gpt2') # use gpt2 weight as pretrained weight
-    example_inputs = (torch.randint(0, 100, (1, 1024), dtype=torch.long), )
+    model = NanoGPT()
+    example_inputs = (torch.randint(0, 100, (1, 3), dtype=torch.long), )
```

Comment:
The reason I update the size here is making the size of input for export the same as the input after export. I think we can enable dynamic export to overcome the issue.